extensions/elasticsearch/elasticsearch-7/src/main/java/com/hazelcast/jet/elastic/ElasticSinkBuilder.java

```diff
@@ -0,0 +1,234 @@
+
+    /**
+     * Set the user-friendly source name for this sink
+     *
+     * @param sinkName user-friendly sink name
+     */
+    @Nonnull
+    public ElasticSinkBuilder<T> name(@Nonnull String sinkName) {
+        this.name = requireNonNull(sinkName, "sinkName");
+        return this;
+    }
+
+    @Nonnull
+    public String name() {
+        return name;
+    }
+
+    /**
+     * Set the client supplier
+     *
+     * @param clientSupplier supplier for configure Elasticsearch REST client
+     */
+    @Nonnull
+    public ElasticSinkBuilder<T> clientSupplier(SupplierEx<? extends RestHighLevelClient> clientSupplier) {
+        checkSerializable(clientSupplier, "clientSupplier");
+        this.clientSupplier = clientSupplier;
+        return this;
+    }
+
+    @Nonnull
+    public SupplierEx<? extends RestHighLevelClient> clientSupplier() {
+        return clientSupplier;
+    }
+
+    /**
+     * Set the destroy function called on completion, defaults to {@link RestHighLevelClient#close()}
+     *
+     * @param destroyFn destroy function
+     */
+    @Nonnull
+    public ElasticSinkBuilder<T> destroyFn(ConsumerEx<? super RestHighLevelClient> destroyFn) {
+        checkSerializable(destroyFn, "destroyFn");
+        this.destroyFn = destroyFn;
+        return this;
+    }
+
+    @Nonnull
+    public ConsumerEx<? super RestHighLevelClient> destroyFn() {
+        return destroyFn;
+    }
+
+    /**
+     * Set the bulkRequestSupplier, defaults to new {@link BulkRequest#BulkRequest()}
+     *
+     * @param bulkRequestSupplier supplier for the bulk request
+     */
+    @Nonnull
+    public ElasticSinkBuilder<T> bulkRequestSupplier(SupplierEx<BulkRequest> bulkRequestSupplier) {
+        checkSerializable(bulkRequestSupplier, "clientSupplier");
+        this.bulkRequestSupplier = bulkRequestSupplier;
+        return this;
+    }
+
+    @Nonnull
+    public SupplierEx<BulkRequest> bulkRequestSupplier() {
+        return bulkRequestSupplier;
+    }
+
+    /**
+     * Set the mapItemFn
+     *
+     * @param mapItemFn maps an item from the stream to an {@link org.elasticsearch.action.index.IndexRequest},
+     *                  {@link org.elasticsearch.action.update.UpdateRequest} or
+     *                  {@link org.elasticsearch.action.delete.DeleteRequest}
+     */
+    @Nonnull
+    public ElasticSinkBuilder<T> mapItemFn(FunctionEx<? super T, ? extends DocWriteRequest<?>> mapItemFn) {
+        checkSerializable(mapItemFn, "mapItemFn");
+        this.mapItemFn = mapItemFn;
+        return this;
+    }
+
+    public FunctionEx<? super T, ? extends DocWriteRequest<?>> mapItemFn() {
+        return mapItemFn;
+    }
+
+    /**
+     * Set the function that provides {@link RequestOptions} based on given request
+     *
+     * @param optionsFn function that provides {@link RequestOptions}
+     */
+    @Nonnull
+    public ElasticSinkBuilder<T> optionsFn(FunctionEx<? super ActionRequest, RequestOptions> optionsFn) {
```

Comment:
this seems something very low level, we shouldn't expose it.

Reply:
The `org.elasticsearch.client.RequestOptions#DEFAULT` may not always work. Users may need to provide custom headers etc..

See
https://www.elastic.co/guide/en/elasticsearch/client/java-rest/current/java-rest-low-usage-requests.html#java-rest-low-usage-request-options